src/pcm/SoxrResampler.cxx

```diff
@@ -80,20 +95,114 @@ soxr_parse_quality(const char *quality) noexcept
 	return SOXR_INVALID_RECIPE;
 }
 
+static unsigned
+SoxrParsePrecision(unsigned value) {
+	switch (value) {
+		case 16:
+		case 20:
+		case 24:
+		case 28:
+		case 32:
+			break;
+		default:
+			throw FormatInvalidArgument(
+				"soxr converter invalid precision : %d [16|20|24|28|32]", value);
+			break;
+	}
+	return value;
+}
+
+static double
+SoxrParsePhaseRespons(unsigned value) {
+	if (value > 100) {
+		throw FormatInvalidArgument(
+			"soxr converter invalid phase_respons : %d (0-100)", value);
+	}
+
+	return double(value);
+}
+
+static double
+SoxrParsePassbandEnd(const char *svalue) {
+	char *endptr;
+	double value = strtod(svalue, &endptr);
+	if (strlen(svalue) == 0 || *endptr) {
+			throw FormatInvalidArgument(
+				"soxr converter passband_end value not a number: %s", svalue);
+	}
+
+	if (value < 1 || value > 100) {
+			throw FormatInvalidArgument(
+				"soxr converter invalid passband_end : %d (1-100%)", value);
```

Comment:
Two bugs in this format string:
- using `%d` but passing a `double`
- unescaped `%`